### ISSUE TITLE
Configuring SessionStateProvider programmatically

### ISSUE THREAD

**Issue by DasOrobas** [Azure/aspnet-redis-providers#34]
Is there a way to configure the SessionStateProvider programmatically ? 

More specifically, I would like to change the host address and the accessKey.

I tried to implement my own SessionModule ([as shown here](https://abhijitjana.net/2011/01/15/programmatically-changing-session-state-behavior-in-asp-net-4-0/)) and getting the SessionStateStoreProviderBase through reflection ([as shown here](https://konste.wordpress.com/2013/01/22/asp-net-accessing-any-session-data-by-session-id/)), but that's cumbersome and I am not sure if that's going to work in the end.

In short, the reason I want to do this, is that I have different environments where I want to deploy my application, and in my case, it would be more convenient to do it like this than with a Web.Config transform. 

Thank you,
Guillaume

**Comment by JasonHaley** [comment]
I'm looking to do the same thing, and it looks like the way to go is to use the connectionString (if you can have different appSettings - which is what i'm looking for).  For you DasOrobas - you may want to look a the settingsClassName and settingsClassMethod - seems like that would allow you to write a utility class that could have whatever host/key per environment you wanted.  More info here: https://github.com/Azure/aspnet-redis-providers/wiki/Configuration

**Comment by SiddharthChatrolaMs** [comment]
@DasOrobas: I hope answer provided by JasonHaley has resolved your issue. Closing this issue for now. if you need other information let us know.
